File: client/src/main/java/com/jraska/github/client/push/PushHandleService.java
package com.jraska.github.client.push;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.jraska.github.client.GitHubClientApp;

public final class PushHandleService extends FirebaseMessagingService {
  @Override public void onMessageReceived(RemoteMessage remoteMessage) {
    GitHubClientApp app = ((GitHubClientApp) getApplication());
    PushAction action = RemoteMessageToActionConverter.convert(remoteMessage);

    app.pushHandler().handlePush(action);
  }
}

File: client/src/main/java/com/jraska/github/client/push/PushHandleService.kt
package com.jraska.github.client.push

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.jraska.github.client.GitHubClientApp

class PushHandleService : FirebaseMessagingService() {
  override fun onMessageReceived(remoteMessage: RemoteMessage) {
    val action = RemoteMessageToActionConverter.convert(remoteMessage)

    pushHandler().handlePush(action)
  }

  override fun onNewToken(p0: String?) {
    pushHandler().onTokenRefresh()
  }

  fun pushHandler(): PushHandler {
    val app = application as GitHubClientApp
    return app.pushHandler()
  }
}
